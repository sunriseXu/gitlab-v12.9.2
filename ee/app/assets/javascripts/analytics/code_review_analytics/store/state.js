export default () => ({
  projectId: null,
  isLoading: false,
  errorCode: null,
  mergeRequests: [],
  pageInfo: {},
  filters: {
    labelName: [],
    milestoneTitle: null,
  },
});
