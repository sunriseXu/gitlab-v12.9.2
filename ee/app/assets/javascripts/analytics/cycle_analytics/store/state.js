import { TASKS_BY_TYPE_SUBJECT_ISSUE } from '../constants';

export default () => ({
  featureFlags: {},

  startDate: null,
  endDate: null,

  isLoading: false,
  isLoadingStage: false,
  isLoadingTasksByTypeChart: false,
  isLoadingDurationChart: false,
  isLoadingDurationChartMedianData: false,

  isEmptyStage: false,
  errorCode: null,

  isSavingCustomStage: false,
  isCreatingCustomStage: false,
  isEditingCustomStage: false,

  selectedGroup: null,
  selectedProjects: [],
  selectedStage: null,

  currentStageEvents: [],

  stages: [],
  summary: [],
  labels: [],
  topRankedLabels: [],
  medians: {},

  customStageFormEvents: [],
  customStageFormErrors: null,
  customStageFormInitialData: null,

  tasksByType: {
    subject: TASKS_BY_TYPE_SUBJECT_ISSUE,
    selectedLabelIds: [],
    data: [],
  },

  durationData: [],
  durationMedianData: [],
});
