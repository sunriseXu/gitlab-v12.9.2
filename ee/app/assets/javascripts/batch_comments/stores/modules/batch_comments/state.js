export default () => ({
  withBatchComments: false,
  isDraftsFetched: false,
  drafts: [],
  isPublishing: false,
  currentlyPublishingDrafts: [],
  isDiscarding: false,
  showPreviewDropdown: false,
});
