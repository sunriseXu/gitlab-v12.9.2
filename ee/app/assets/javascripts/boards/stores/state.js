import createStateCE from '~/boards/stores/state';

export default () => ({
  ...createStateCE(),

  // ...
});
