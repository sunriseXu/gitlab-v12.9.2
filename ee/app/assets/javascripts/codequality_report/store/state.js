export default () => ({
  endpoint: '',
  allCodequalityIssues: [],
  isLoadingCodequality: false,
  loadingCodequalityFailed: false,
  codeQualityError: null,
  pageInfo: {
    page: 1,
    perPage: 25,
    total: 0,
  },
});
