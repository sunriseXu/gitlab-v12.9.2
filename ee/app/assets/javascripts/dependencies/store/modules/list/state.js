import { FILTER, REPORT_STATUS, SORT_ORDER } from './constants';

export default () => ({
  endpoint: '',
  initialized: false,
  isLoading: false,
  errorLoading: false,
  dependencies: [],
  pageInfo: {
    total: 0,
  },
  reportInfo: {
    status: REPORT_STATUS.ok,
    jobPath: '',
    generatedAt: '',
  },
  filter: FILTER.all,
  sortField: 'name',
  sortOrder: SORT_ORDER.ascending,
});
