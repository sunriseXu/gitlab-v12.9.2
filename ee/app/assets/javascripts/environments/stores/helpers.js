/**
 * Deploy boards are EE only.
 *
 * @param {Object} environment
 * @returns {Object}
 */
// eslint-disable-next-line import/prefer-default-export
export const setDeployBoard = (oldEnvironmentState, environment) => {
  let parsedEnvironment = environment;
  if (environment.size === 1 && environment.rollout_status) {
    parsedEnvironment = Object.assign({}, environment, {
      hasDeployBoard: true,
      isDeployBoardVisible:
        oldEnvironmentState.isDeployBoardVisible === false
          ? oldEnvironmentState.isDeployBoardVisible
          : true,
      deployBoardData:
        environment.rollout_status.status === 'found' ? environment.rollout_status : {},
      isLoadingDeployBoard: environment.rollout_status.status === 'loading',
      isEmptyDeployBoard: environment.rollout_status.status === 'not_found',
      hasLegacyAppLabel: environment.rollout_status.has_legacy_app_label,
    });
  }
  return parsedEnvironment;
};
