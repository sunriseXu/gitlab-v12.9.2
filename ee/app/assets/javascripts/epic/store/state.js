export default () => ({
  // API Paths to Send/Receive Data
  endpoint: '',
  updateEndpoint: '',

  fullPath: '',
  groupPath: '',
  markdownPreviewPath: '',
  labelsPath: '',
  todoPath: '',
  todoDeletePath: '',

  // URLs to use with links
  epicsWebUrl: '',
  labelsWebUrl: '',
  markdownDocsPath: '',

  // Flags
  canUpdate: false,
  canDestroy: false,
  canAdmin: false,
  allowSubEpics: false,

  // Epic Information
  epicId: 0,
  namespace: '#',
  state: '',
  created: '',
  author: null,
  initialTitleHtml: '',
  initialTitleText: '',
  initialDescriptionHtml: '',
  initialDescriptionText: '',
  lockVersion: 0,

  todoExists: false,
  startDateSourcingMilestoneTitle: '',
  startDateSourcingMilestoneDates: {
    startDate: '',
    dueDate: '',
  },
  startDateIsFixed: false,
  startDateFixed: '',
  startDateFromMilestones: '',
  startDate: '',
  dueDateSourcingMilestoneTitle: '',
  dueDateSourcingMilestoneDates: {
    startDate: '',
    dueDate: '',
  },
  dueDateIsFixed: '',
  dueDateFixed: '',
  dueDateFromMilestones: '',
  dueDate: '',
  labels: [],
  ancestors: [],
  participants: [],
  subscribed: false,

  // Create Epic Props
  newEpicTitle: '',

  // UI status flags
  epicStatusChangeInProgress: false,
  epicDeleteInProgress: false,
  epicTodoToggleInProgress: false,
  epicStartDateSaveInProgress: false,
  epicDueDateSaveInProgress: false,
  epicLabelsSelectInProgress: false,
  epicSubscriptionToggleInProgress: false,
  epicCreateInProgress: false,
  sidebarCollapsed: false,
});
