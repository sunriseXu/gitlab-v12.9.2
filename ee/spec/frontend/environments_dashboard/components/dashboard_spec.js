import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import { GlButton, GlModal, GlSprintf, GlLink } from '@gitlab/ui';
import createStore from 'ee/vue_shared/dashboards/store/index';
import state from 'ee/vue_shared/dashboards/store/state';
import component from 'ee/environments_dashboard/components/dashboard/dashboard.vue';
import ProjectHeader from 'ee/environments_dashboard/components/dashboard/project_header.vue';
import Environment from 'ee/environments_dashboard/components/dashboard/environment.vue';
import ProjectSelector from '~/vue_shared/components/project_selector/project_selector.vue';
import { trimText } from 'helpers/text_helper';

import environment from './mock_environment.json';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('dashboard', () => {
  let actionSpies;
  const store = createStore();
  let wrapper;
  let propsData;

  beforeEach(() => {
    actionSpies = {
      addProjectsToDashboard: jest.fn(),
      clearSearchResults: jest.fn(),
      setSearchQuery: jest.fn(),
      fetchSearchResults: jest.fn(),
      removeProject: jest.fn(),
      toggleSelectedProject: jest.fn(),
      fetchNextPage: jest.fn(),
      fetchProjects: jest.fn(),
    };
    propsData = {
      addPath: 'mock-addPath',
      listPath: 'mock-listPath',
      emptyDashboardSvgPath: '/assets/illustrations/operations-dashboard_empty.svg',
      emptyDashboardHelpPath: '/help/user/operations_dashboard/index.html',
      environmentsDashboardHelpPath: '/help/user/operations_dashboard/index.html',
    };

    wrapper = shallowMount(component, {
      propsData,
      localVue,
      store,
      methods: {
        ...actionSpies,
      },
      stubs: { GlSprintf },
    });
  });

  afterEach(() => {
    wrapper.destroy();
    store.replaceState(state());
  });

  it('should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders the dashboard title', () => {
    expect(wrapper.find('.js-dashboard-title').text()).toBe('Environments Dashboard');
  });

  describe('page limits information message', () => {
    let message;

    beforeEach(() => {
      message = wrapper.find('.js-page-limits-message');
    });

    it('renders the message', () => {
      expect(trimText(message.text())).toBe(
        'This dashboard displays a maximum of 7 projects and 3 environments per project. Read more.',
      );
    });

    it('includes the correct documentation link in the message', () => {
      const helpLink = message.find(GlLink);

      expect(helpLink.text()).toBe('Read more.');
      expect(helpLink.attributes('href')).toBe(propsData.environmentsDashboardHelpPath);
      expect(helpLink.attributes('rel')).toBe('noopener noreferrer');
    });
  });

  describe('add projects button', () => {
    let button;

    beforeEach(() => {
      button = wrapper.find(GlButton);
    });

    it('is labelled correctly', () => {
      expect(button.text()).toBe('Add projects');
    });
  });

  describe('wrapped components', () => {
    beforeEach(() => {
      store.state.projects = [
        {
          id: 0,
          name: 'test',
          namespace: { name: 'test', id: 0 },
          environments: [{ ...environment, id: 0 }, environment],
        },
        { id: 1, name: 'test', namespace: { name: 'test', id: 0 }, environments: [environment] },
      ];
    });

    describe('project header', () => {
      it('should have one project header per project', () => {
        const headers = wrapper.findAll(ProjectHeader);
        expect(headers.length).toBe(2);
      });

      it('should remove a project if it emits `remove`', () => {
        const header = wrapper.find(ProjectHeader);
        header.vm.$emit('remove');
        expect(actionSpies.removeProject).toHaveBeenCalled();
      });
    });

    describe('environment component', () => {
      it('should have one environment component per environment', () => {
        const environments = wrapper.findAll(Environment);
        expect(environments.length).toBe(3);
      });
    });

    describe('project selector modal', () => {
      beforeEach(() => {
        wrapper.find(GlButton).trigger('click');
        return wrapper.vm.$nextTick();
      });

      it('should fire the add projects action on ok', () => {
        wrapper.find(GlModal).vm.$emit('ok');
        expect(actionSpies.addProjectsToDashboard).toHaveBeenCalled();
      });

      it('should fire clear search when the modal is hidden', () => {
        wrapper.find(GlModal).vm.$emit('hidden');
        expect(actionSpies.clearSearchResults).toHaveBeenCalled();
      });

      it('should set the search query when searching', () => {
        wrapper.find(ProjectSelector).vm.$emit('searched', 'test');
        expect(actionSpies.setSearchQuery).toHaveBeenCalledWith('test');
      });

      it('should fetch query results when searching', () => {
        wrapper.find(ProjectSelector).vm.$emit('searched', 'test');
        expect(actionSpies.fetchSearchResults).toHaveBeenCalled();
      });

      it('should toggle a project when clicked', () => {
        wrapper.find(ProjectSelector).vm.$emit('projectClicked', { name: 'test', id: 1 });
        expect(actionSpies.toggleSelectedProject).toHaveBeenCalledWith({ name: 'test', id: 1 });
      });

      it('should fetch the next page when bottom is reached', () => {
        wrapper.find(ProjectSelector).vm.$emit('bottomReached');
        expect(actionSpies.fetchNextPage).toHaveBeenCalled();
      });

      it('should get the page info from the state', () => {
        store.state.pageInfo = { totalResults: 100 };

        return wrapper.vm.$nextTick().then(() => {
          expect(wrapper.find(ProjectSelector).props('totalResults')).toBe(100);
        });
      });
    });
  });
});
