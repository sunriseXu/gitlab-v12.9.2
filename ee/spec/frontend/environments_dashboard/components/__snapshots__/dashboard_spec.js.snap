// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard should match the snapshot 1`] = `
<div
  class="operations-dashboard"
>
  <gl-modal-stub
    modalclass=""
    modalid="add-projects-modal"
    ok-disabled="true"
    ok-title="Add projects"
    ok-variant="success"
    size="md"
    title="Add projects"
    titletag="h4"
  >
    <project-selector-stub
      projectsearchresults=""
      selectedprojects=""
      totalresults="0"
    />
  </gl-modal-stub>
   
  <div
    class="page-title-holder flex-fill d-flex align-items-center"
  >
    <h1
      class="js-dashboard-title page-title text-nowrap flex-fill"
    >
      
      Environments Dashboard
    
    </h1>
     
    <gl-button-stub
      class="js-add-projects-button btn btn-success"
      role="button"
      size="md"
      tabindex="0"
      variant="secondary"
    >
      
      Add projects
    
    </gl-button-stub>
  </div>
   
  <p
    class="mt-2 mb-4 js-page-limits-message"
  >
    This dashboard displays a maximum of 7 projects and 3 environments per project. 
    <gl-link-stub
      href="/help/user/operations_dashboard/index.html"
      rel="noopener noreferrer"
      target="_blank"
    >
      
          Read more.
        
    </gl-link-stub>
  </p>
   
  <div
    class="prepend-top-default"
  >
    <div
      class="row prepend-top-20 text-center"
    >
      <div
        class="col-12 d-flex justify-content-center svg-content"
      >
        <img
          class="js-empty-state-svg col-12 prepend-top-20"
          src="/assets/illustrations/operations-dashboard_empty.svg"
        />
      </div>
       
      <h4
        class="js-title col-12 prepend-top-20"
      >
        
        Add a project to the dashboard
      
      </h4>
       
      <div
        class="col-12 d-flex justify-content-center"
      >
        <span
          class="js-sub-title mw-460 text-tertiary text-left"
        >
          
          The environments dashboard provides a summary of each project's environments' status, including pipeline and alert statuses.
        
        </span>
      </div>
       
      <div
        class="col-12"
      >
        <a
          class="js-documentation-link btn btn-primary prepend-top-default append-bottom-default"
          href="/help/user/operations_dashboard/index.html"
        >
          
          View documentation
        
        </a>
      </div>
    </div>
  </div>
</div>
`;
